shopinvader_api_quotation/models/product_template.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2017-2018 Akretion (http://www.akretion.com).
+# Copyright 2021 Camptocamp (https://www.camptocamp.com).
+# @author Benoît GUILLOT <benoit.guillot@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    shop_only_quotation = fields.Boolean(
```

Comment:
On V16 the following code will not work anymore (due to some ORM cleanup)
We should replace the boolean on the product_template by a "selection field" with the following value

```
   ("all_variant", "All Variant"),
   ("manually_on_variant", "Manually on Variant"),
   ("never", "Never"),
```


Then the inverse method will do the following

all_variant => all variant to True
manually_on_variant => do nothing
never => all variant to False

The compute method should do the following
all variant to True => all_variant
mix => manually_on_variant
all variant to False => never

So doing a recompute will be stable and the test will be not brokend anymore

Reply:
we should also rename the field to "shop_only_quotation_mode" so we do not have "name" conflict between product and template